### PR TITLE
Add vagrant setup

### PR DESCRIPTION
**Pull request by zhenik**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,4 +11,8 @@ endif
 build: custom_ca
 	docker build . -t local/hive:$(branch)
 	docker tag  local/hive:$(branch) local/hive:latest
+up:
+	vagrant up --provision
```
Comment (fredrikhgrelland) [inline Makefile:15]: -- Is this what we want? vagrant up should be sufficient. --provision will cause it to always run the provisioning.

File: test/hive-connect.hcl [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+      lifecycle {
+        hook    = "prestart"
+      }
+
+      driver = "docker"
+      config {
+        image = "alioygur/wait-for:latest"
+        args = [ "-it", "${NOMAD_UPSTREAM_ADDR_minio}", "-t", 120 ]
+      }
+    }
```
Comment (fredrikhgrelland) [inline test/hive-connect.hcl:209]: We should change this to something that actually checks minio, no?

File: Makefile (hunk 1 above)
Comment (zhenik) [reply Makefile:15]: Yes. 

ps. I do not want to mix issue #4 in this PR. Next PR I will focus on CD/CI and how to separate provision by some input for local `setup` or `test`.

File: test/hive-connect.hcl (hunk 2 above)
Comment (zhenik) [reply test/hive-connect.hcl:209]: Yeah, I have an idea to build REST service which query consul by (e.g.) service name and verify that all health checks for target service are green, then exit:0 . But it is also a separate issue.

File: Makefile (hunk 1 above)
Comment (fredrikhgrelland) [reply Makefile:15]: yup

File: test/hive-connect.hcl (hunk 2 above)
Comment (fredrikhgrelland) [reply test/hive-connect.hcl:209]: @zhenik I'll have a quick stab at it.

**Pull request merged**